var/spack/repos/builtin.mock/packages/attributes-foo/package.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class AttributesFoo(BundlePackage):
+    phases = ['install']
+    version('1.0')
+
+    variant('bar', default=False, description='Enable the Foo implementation of bar')
+    variant('baz', default=False, description='Enable the Foo implementation of baz')
+
+    provides('bar', when='+bar')
+    provides('baz', when='+baz')
+
+    def install(self, spec, prefix):
+        def libname(basename):
+            _prefix = '' if 'platform=windows' in spec else 'lib'
+            if 'platform=windows' in spec:
+                _suffix = '.lib'
+            elif 'platform=darwin' in spec:
+                _suffix = '.dylib'
+            else:
+                _suffix = '.so'
+            return _prefix + basename + _suffix
+
+        mkdirp(prefix.join('include'))
+        touch(prefix.join('include/foo.h'))
+        mkdirp(prefix.join('include/bar'))
+        touch(prefix.join('include/bar/bar.h'))
+        mkdirp(prefix.join('lib64'))
+        touch(prefix.join('lib64/{}'.format(libname('Foo'))))
+        touch(prefix.join('lib64/{}'.format(libname('FooBar'))))
+        mkdirp(prefix.join('baz/include/baz'))
+        touch(prefix.join('baz/include/baz/baz.h'))
+        mkdirp(prefix.join('baz/lib'))
+        touch(prefix.join('baz/lib/{}'.format(libname('FooBaz'))))
```

Comment:
The layout is to specifically cover the use case of the `baz` virtual package being installed in a subdirectory of of the prefix instead of at the top level of the prefix.